crates/controller/src/actions.rs

```diff
@@ -0,0 +1,115 @@
+use bevy::input::keyboard::KeyCode;
+use bevy::prelude::Reflect;
+use bevy::prelude::{MouseButton, Res};
+use de_input::{AppKeybinding, DefaultKeybindings};
+use leafwing_input_manager::prelude::{ActionState, InputMap, UserInput};
+use leafwing_input_manager::Actionlike;
+use serde::{Deserialize, Serialize};
+
+pub struct ActionPlugin;
+
+impl bevy::app::Plugin for ActionPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_action_set::<Action>("actions")
+            // Mouse is separate because otherwise it will clash with AddToSelection and ReplaceSelection
+            .add_action_set::<MouseAction>("mouse_actions");
+    }
+}
+
+/// make actoinlike enum that has normal actions and factory actions. an action is A vareint followed by a KeyConfig<A>
+macro_rules! make_actions {
```

Comment:
I would get rid of this macro and implement all the stuff directly. It is used only once so IMHO it is an unnecessary indirection.